v2/cmd/controller/main.go

```diff
@@ -197,3 +303,96 @@ func makeControllerOptions(log logr.Logger, cfg config.Values) generic.Options {
 			}),
 	}
 }
+
+func getCRDsToSkip(ctx context.Context, logger logr.Logger, k8sConfig *rest.Config, cfg config.Values) (map[string]apiextensions.CustomResourceDefinition, error) {
```

Comment:
Changed this to `getReadyCRDs`. I support both now in the `crds` library as well (just invert the comparators and call a single impl). I think this solves your issue and is a bit cleaner.

Reply:
Actually upon further investigation, there is a downside to doing it this way:

1. controller_gen.go methods return the set of all resources we understand (GetKnownTypes). This includes `InstalledResourceDefinitions` (because we understand it).
2. `crdManager.LoadOperatorCRDs` returns the set of all CRDs the operator itself is managing. This includes all CRDs _except_ the `InstalledResourceDefinitions` CRD. That CRD is special because the operator doesn't manage it.
3. `crdManager.ListOperatorCRDs` returns the set of all CRDs that the operator finds on APIServer. In the happy path flow this is first _just_ `InstalledResourceDefinitions`, and then after the `InstalledResourceDefinitions` resource is reconciled it's all CRDs.

We effectively are defining the following sets of things:

```
crdsToReconcile = crdManager.LoadOperatorCRDs ∩ crdManager.ListOperatorCRDs
crdsToInstall = crdManager.LoadOperatorCRDs \ crdManager.ListOperatorCRDs
clusterScopeCRDs = controllers.GetClusterScopeStorageTypes
crdsToRunWebhooksOn = ???
```

```
∩ = set intersection
∪  = set union
\ = set difference
```

In order to correctly generate the set of types to run webhooks on we want:
`crdsToReconcile ∪ clusterScopeCRDs` - but the types here don't actually line up, so the easier way to produce this set is actually `controllers.GetKnownTypes \ crdsToInstall`

`crdsToInstall` was the `skip` I had before.

I've kept the other changes but moved this back to a `skip` (set difference) approach because of the above.